src/main/java/io/temporal/samples/interceptor/collector/CountCollector.java

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.interceptor.collector;
+
+import io.temporal.activity.ActivityExecutionContext;
+import io.temporal.workflow.WorkflowInfo;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CountCollector {
```

Comment:
CountCollector/it's internal collectons have to be thread-safe.
`InterceptorStarter.interceptor.getCountCollector().getSignalsInfoList().add(signalCollector);`
This code is/can be executed in several concurrent threads and they share the same one `CountCollector`.

Reply:
Yes, its just sample but will try to change to the comments.